universal-adapter-kotlin/src/main/java/com/appunite/universal_adapter_kotlin/UniversalAdapter.kt

```diff
@@ -0,0 +1,70 @@
+package com.appunite.universal_adapter_kotlin
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import com.jacekmarchwicki.universaladapter.BaseAdapterItem
+
+class UniversalAdapter(
```

Comment:
Can't you just use the java version of the UniversalAdapter so two implementations don't need to be maintained?

Reply:
That one is tricky. Kotlin version inherits from androidx.ListAdapter which handles list item changes with DiffUtil.
On the other hand, the Java version uses RecyclerView.Adapter and implements ChangesDetector.ChangesAdapter.

There is some common code that could be re-used with some common interface and delegation but the universal-adapter-kotlin would have to depend on universal-adapter module. So we'd unnecessarily have changes detector stuff in the universal-adapter-kotlin module. More to that it's getting complicated. 
Do you have any idea how to approach this?